test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/Parallel/ParallelProxyDiscoveryManagerTests.cs

```diff
@@ -41,102 +43,114 @@ public ParallelProxyDiscoveryManagerTests()
             AssertMissingAndDuplicateSources(processedSources);
         }
 
         /// <summary>
         ///  Create ParallelProxyDiscoveryManager with parallel level 1 and two source,
-        ///  Abort in any source should stop discovery for other sources.
+        ///  Abort in any source should not stop discovery for other sources.
         /// </summary>
         [TestMethod]
         public void DiscoveryTestsShouldProcessAllSourcesOnDiscoveryAbortsForAnySource()
         {
             var discoveryManagerMock = new Mock<IProxyDiscoveryManager>();
-            proxyParallelDiscoveryManager = new ParallelProxyDiscoveryManager(() => discoveryManagerMock.Object, 1, false);
             this.createdMockManagers.Add(discoveryManagerMock);
-            var processedSources = new List<string>();
-            SetupDiscoveryTests(processedSources, true);
-            AutoResetEvent completeEvent = new AutoResetEvent(false);
-            SetupHandleDiscoveyComplete(completeEvent, true);
+            var parallelDiscoveryManager = this.SetupDiscoveryManager(() => discoveryManagerMock.Object, 1, false);
 
             Task.Run(() =>
             {
-                proxyParallelDiscoveryManager.DiscoverTests(this.testDiscoveryCriteria, mockHandler.Object);
+                parallelDiscoveryManager.DiscoverTests(this.testDiscoveryCriteria, this.mockHandler.Object);
             });
 
-            Assert.IsTrue(completeEvent.WaitOne(ParallelProxyDiscoveryManagerTests.taskTimeout), "Test discovery not completed.");
-            Assert.AreEqual(2, processedSources.Count, "All Sources must not be processed.");
+            Assert.IsTrue(this.discoveryCompleted.Wait(ParallelProxyDiscoveryManagerTests.taskTimeout), "Test discovery not completed.");
+            Assert.AreEqual(2, processedSources.Count, "All Sources must be processed.");
         }
 
         [TestMethod]
-        public void HandlePartialDiscoveryCompleteShouldCreateANewProxyDiscoveryManagerIfIsAbortedIsTrue()
+        public void DiscoveryTestsShouldProcessAllSourceIfOneDiscoveryManagerIsStarved()
         {
-            proxyParallelDiscoveryManager = new ParallelProxyDiscoveryManager(this.proxyManagerFunc, 1, false);
+            // Ensure that second discovery manager never starts. Expect 10 total tests.
+            var parallelDiscoveryManager = this.SetupDiscoveryManager(this.proxyManagerFunc, 2, true, totalTests: 10);
+            this.createdMockManagers[1].Reset();
+            this.createdMockManagers[1].Setup(dm => dm.DiscoverTests(It.IsAny<DiscoveryCriteria>(), It.IsAny<ITestDiscoveryEventsHandler>()))
+                .Throws<NotImplementedException>();
 
+            Task.Run(() =>
+            {
+                parallelDiscoveryManager.DiscoverTests(this.testDiscoveryCriteria, this.mockHandler.Object);
+            });
+
+            // Processed sources should be 1 since the 2nd source is never discovered
```

Comment:
Nit: keep comment below one line and update msg to something like "Only one source must be processed".